app/src/main/java/com/android/solvit/shared/ui/map/MapScreen.kt

```diff
@@ -80,13 +104,38 @@ fun MapScreen(userLocation: LatLng?, markers: List<MarkerData>, bottomBar: @Comp
             verticalArrangement = Arrangement.Center,
             horizontalAlignment = Alignment.CenterHorizontally,
         ) {
-          MapContent(userLocation = userLocation, markers = markers)
+          Box {
+            MapContent(userLocation = userLocation, markers = markers)
+            if (markersLoading) {
+              Column(modifier = Modifier.align(Alignment.TopCenter).padding(16.dp)) {
+                OutlinedCard(
+                    shape = RoundedCornerShape(32.dp),
+                    elevation = CardDefaults.outlinedCardElevation()) {
+                      Row(
+                          modifier = Modifier.padding(8.dp),
+                          verticalAlignment = Alignment.CenterVertically) {
+                            Text("Loading images  ", fontWeight = FontWeight.Bold)
+                            CircularProgressIndicator(
+                                modifier = Modifier.size(30.dp), color = colorScheme.onBackground)
+                          }
+                    }
+              }
```

Comment:
You could animate the loading a little bit more with an Animated Visibility
```
AnimatedVisibility(
    visible = markersLoading,
    enter = fadeIn() + slideInVertically(),
    exit = fadeOut() + slideOutVertically()
) {
    Column(
        modifier = Modifier
            .align(Alignment.TopCenter)
            .padding(16.dp)
            .testTag("loading_indicator")
    ) {
        OutlinedCard(
            shape = RoundedCornerShape(32.dp),
            elevation = CardDefaults.outlinedCardElevation()
        ) {
            Row(
                modifier = Modifier.padding(8.dp),
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(8.dp)
            ) {
                Text(
                    "Loading images",
                    fontWeight = FontWeight.Bold,
                    style = MaterialTheme.typography.labelLarge
                )
                CircularProgressIndicator(
                    modifier = Modifier.size(24.dp),
                    color = colorScheme.primary,
                    strokeWidth = 2.dp
                )
            }
        }
    }
}
```

Reply:
Actually `AnimatedVisibility` cannot be used inside a `Box` composable, what a shame! this would have been very cool